sklearn_numba_dpex/kmeans/kernels/lloyd_single_step.py

```diff
@@ -334,9 +361,7 @@ def fused_lloyd_single_step(
         # counts.
 
         # each work item is assigned an array of centroids in a round robin manner
-        privatization_idx = (
-            sample_idx // sub_group_size
-        ) % n_centroids_private_copies
+        privatization_idx = sub_group_idx % n_centroids_private_copies
```

Comment:
Is it possible to avoid using a modulo, here?

Reply:
I don't think so :-(